src/configure/clients/github/githubClient.ts

```diff
@@ -9,61 +10,113 @@ export class GithubClient {
-        let encryptedEncodedText = SodiumLibHelper.encodeToBase64(encryptedBytesAsString);
-        await this._setGithubSecret(secretName, secretKeyObject.key_id, encryptedEncodedText);
+    public async createGithubRepo(orgName: string, repoName: string): Promise<void | GitHubRepo> {
+        let restClient =new RestClient();
+        let Url = "https://api.github.com/orgs/" + orgName + "/repos";
+        try{
+            return restClient.sendRequest(<UrlBasedRequestPrepareOptions>{
+                url: Url,
+                headers: {
+                    "Content-Type": "application/json",
+                    "Authorization": "Bearer " + this.patToken,
+                    "User-Agent": UserAgent
+                },
+                method: 'POST',
+                body: {
+                    "name": repoName,
+                    "description": "Repo created from VScode extension 'Deploy to Azure'",
+                    "homepage": "https://github.com",
+                    "private": false,
+                    "has_issues": true,
+                    "has_projects": true,
+                    "has_wiki": true
+                },
+                deserializationMapper: null,
+                serializationMapper: null
+            })
+                .then((detail: GitHubRepo) => {
+                    return detail;
+                }).catch(error=>{
+                        return null;
+                });
+        }
+        catch(error){
+            return null;
+        }
     }
 
-    private async _getGitHubSecretKey() : Promise<GitHubSecretKey> {
-        let request = <UrlBasedRequestPrepareOptions> {
-                url: GitHubProvider.getFormattedGitHubApiUrlBase(this.url) + "/actions/secrets/public-key",
+    public async listOrganizations(forceRefresh?: boolean): Promise<void | GitHubOrganization[]> {
+        if (!this.listOrgPromise || forceRefresh) {
+            let restClient = new RestClient();
+            this.listOrgPromise = restClient.sendRequest(<UrlBasedRequestPrepareOptions>{
+                url: "https://api.github.com/user/orgs",
                 method: 'GET',
                 headers: {
+                    "Authorization": "Bearer " + this.patToken,
                     "User-Agent": UserAgent,
                     "Content-Type": "application/json",
-                    "Authorization": "Bearer " + this.patToken,
                     "Accept": "*/*"
                 },
-                serializationMapper: null,
-                deserializationMapper: null
-            }
-        let restClient = new RestClient();
-        return (await restClient.sendRequest(request)) as GitHubSecretKey;
+                deserializationMapper: null,
+                serializationMapper: null
+            })
+                .then((organizations: Array<GitHubOrganization>) => {
+                    organizations = organizations.sort((org1, org2) => stringCompareFunction(org1.login, org2.login));
```

Comment:
Is there a way to get only those orgs where user has create repo permission?

Reply:
No, none of the GitHub APIs provides this feature.